doc/tech/approach.md

```diff
@@ -3,97 +3,226 @@
+
+•	**Workload diversity**: Should be capable to handle any VNF , CNF and BareMetal workloads.
+
+•	**Telco Grade**: Should support deployments with Telco Grade characteristics requirements.
+
+•	**Future proof**: Should support known and emerging technology trends e.g. SmartNICs, FPGAs and Programmable Switches, multi-cloud aspects (public/private/hybrid), multi-location aspects (multi-cluster, DCI, etc), Edge realted technologies, etc.
+
+
+<a name="5.1.3"></a> 
+### 5.1.3 Network Layering and Concepts
+
+Cloud and Telco networking are layered, and it is very important to keep the layering dependencies low to enable security, separation and portability in between multiple implementations.
+
+Before we start developing a deep model we need to agree on some foundational concepts and layering that allow decoupling of implementations in between the layers. We will emphasize four concepts in this section
+
+ - Underlay and Overlay Networking concepts
+ - Hardware and Software Infrastructure Layer concepts
+ - Software Defined Networking, SDNu and SDNo concepts
+ - Programmable Networking Fabric concept
+
+<a name="5.1.3.1"></a>
+#### 5.1.3.1 Underlay and Overlay Networking concepts
+
+The ETSI NFV model divides networking into an Underlay and an Overlay Network layer. The purpose with this layering is to ensure separation of the SW Virtualisation tenants (Workload) Overlay Networks from each other, whilst allowing the traffic to flow on the shared Underlay Network in between all Ethernet connected HW units.
+
+The Overlay Networking separation is often done through encapsulation of Tenants' traffic using overlay protocols e.g. through VxLAN or EVPN over the Underlay Network e.g. based on L2 (VLAN) or L3 (IP) networks.
+
+In some instances, the SW Virtualisation Tenants can bypass the Overlay Networking encapsulation to achieve better performance or network visibility/control. A common method to bypass the Overlay Networking encapsulation normally done by the SW Virtualisation Layer, is the VNF/CNF usage of SR-IOV that effectively take over the Physical and Virtual Functions of the NIC directly into the VNF/CNF Tenant. In these cases, the Underlay Networking must handle the separation e.g. through a Virtual Termination End Point (VTEP) that encapsulate the Overlay Network traffic.
+
+> **Note:** Bypassing the Overlay Networking layer is a violation of the basic CNTT decoupling principles, but in some cases unavoidable with existing technologies and standards. Until suitable technologies and standards are developed, CNTT have a set of agreed exemptions that forces the Underlay Networking to handle the bypassed Overlay Networking separation.
+
+VTEP could be manually provisioned in the Underlay Networking or be automated and controlled through a Software Defined Networking controller interfaces into the underlying networking in the HW Infrastructure Layer. 
```

Comment:
Based on painful experience, I'd strongly advise against CNTT supporting manual manipulation of the underlay fabric to effect support of SRIOV. In production it turns into an Operations **nightmare**, and is a recipe for outages. I can't stress this enough. @ASawwaf @TFredberg @ulikleber Thoughts?

Reply:
@markshostak I fully agree that manual configurations e.g. of VTEP and mapping onto the underlay fabric is painful, but that is where some operator deployments end up if they have a traditional Networking organization that owns the management/administration of the underlay switches and the Cloud Infrastructure organization that owns/use the servers have to deploy performance critical applications that bypass the overlay switching.
I am though not sure if CNTT can forbid manual manipulation.
Is there possibly some sort of CNTT Advisory Guidelines or Warnings that could express this, or should we just try to build in these warnings into all places where we think it is a bad practice?

Reply:
@TFredberg 
>
> I am though not sure if CNTT can forbid manual manipulation.

An Operator can do whatever they like; we don't _forbid_ anything. However, that doesn't mean that CNTT has to support it. Further, there are cases where CNTT can go farther than not supporting, but actually recommend against a certain practice (as I believe is appropriate in this case). Basically, some people are going to go to the beach, and you just can't stop them... :-\

Reply:
Good discussion, @markshostak and @TFredberg. I have changed to "VTEP could be manually provisioned in the Underlay Networking, which is not a recommended practice, or be automated and controlled through a Software Defined Networking controller interfaces into the underlying networking in the HW Infrastructure Layer. " It is more neutral than forbidding, but it is a clear message.  I do not think that many people would to go manual with this.  I have some painful memories as well...